src/index.ts

```diff
@@ -59,6 +60,12 @@ export class Datastore {
     async addFile(name: string, publicStatus: boolean, file: ArrayBuffer) {
         await this._initialize()
 
+        if (JSZip.support.arraybuffer) {
+            let zip = JSZip()
+            await zip.file(name, file)
+            file = await zip.generateAsync({type : "arraybuffer"})
+        }
+
         let encryptionKey = ""
         let storageId
         if (!publicStatus) {
```

Comment:
I can't comment on lines 75 and 78. What do you think if we use the `byteLength` of the uncompressed file instead?

Reply:
Yup! Will do